pkg/containermanager/docker/docker_manager.go

```diff
@@ -80,6 +80,20 @@ func (dc dockerContainerManager) Provision(ctx context.Context, containerConfig
 	return nil
 }
 
+func (dc dockerContainerManager) Deprovision(ctx context.Context, containerConfig *containermanager.ContainerConfig) error {
+	err := dc.client.ContainerStop(ctx, containerConfig.Name, nil)
+	if err != nil {
+		dc.logger.Error("container-stopping-failed", err)
+		return err
+	}
+	err = dc.client.ContainerRemove(ctx, containerConfig.Name, types.ContainerRemoveOptions{})
```

Comment:
I think we should set `RemoveOptions{Force:true}`. No real valid reason to fail on going away.

Reply:
let's force the removal but for now we can break out with the error if it fails to stop or remove.

Reply:
force remove is kill. I figure less traffic back and forth is less things to fail.

Reply:
Which way do we want to go? 
Do we want to gracefully stop and remove as Jonathan is doing, and return errors when it fails so the PaaS is notified about the failure to deprovision.
OR 
Do we want to just kill it every time? `client.ContainerKill`

Reply:
I like the graceful option, because then if something is actually wrong people would be notified. And this is a unlikely use case to begin with so if we are thinking about that, I think it makes more sense for us say deprovision failed.

Reply:
kill is not remove.

Reply:
i would say lets bubble up the error for stop but force remove.

Reply:
i don't think forcing remove is a good idea. Pretty sure if it fails to term, we DONT wan't to kill it, because it's probably not terminating for a good reason.

Reply:
okay, then we need to be careful with what we send for stop, and possibly send a kill instead.

Reply:
To be specific, use the Docker Kill command (which ought better to be called Docker Signal, and send exactly the signal we want to be sending, (which I'm not sure how we know what that is, but I believe it is configurable on the container itself).